SocialCareCaseViewerApi/V1/Gateways/ProcessDataGateway.cs

```diff
@@ -38,11 +39,29 @@ public IEnumerable<CareCaseData> GetProcessData(ListCasesRequest request)
                     throw new Exception();
                 }
             }
-            var filter = !string.IsNullOrEmpty(request.WorkerEmail) ?
-                Builders<BsonDocument>.Filter.Eq("worker_email", request.WorkerEmail)
-                : "{$or: [{ mosaic_id: " + mosaicId.ToString() + "}, { mosaic_id: /" + mosaicId.ToString() + "/}]}";
+            var officerEmailFilter = !string.IsNullOrWhiteSpace(request.WorkerEmail) ?
+                Builders<BsonDocument>.Filter.Eq("worker_email", request.WorkerEmail) : null;
+            var mosaicIdFilter = !string.IsNullOrWhiteSpace(mosaicId.ToString()) ?
```

Comment:
check whether mosaicId is null before attempting to convert it to string

Reply:
Updated